src/components/App.tsx

```diff
@@ -0,0 +1,61 @@
+import { RecordOf } from 'immutable';
+
+import React, { FC, useContext, useEffect } from 'react';
+
+import { Context, LocalContext } from '@graasp/apps-query-client';
+import { PermissionLevel } from '@graasp/sdk';
+
+import { DEFAULT_CONTEXT_LANGUAGE } from '../config/appSettings';
+import i18n from '../config/i18n';
+import { DEFAULT_PERMISSION } from '../config/settings';
+import '../index.css';
+import { AppDataProvider } from './context/AppDataContext';
+import { AppSettingProvider } from './context/AppSettingContext';
+import { CanvasProvider } from './context/CanvasContext';
+import { MembersProvider } from './context/MembersContext';
+import StudentView from './modes/student/StudentView';
+import TeacherView from './modes/teacher/TeacherView';
+
+const App: FC = () => {
+  // context describes the item context, i.e. has the item id, current member id (memberId),
+  // the language and current view (builder, player, ...), the current permission (admin, write, read)
+  const context: RecordOf<LocalContext> = useContext(Context);
+
+  const permissionLevel =
+    (context?.get('permission', DEFAULT_PERMISSION) as PermissionLevel) ||
```

Comment:
Is the casting mandatory here ? Maybe you can use:
```ts
const permissionLevel: PermissionLevel = context?.get('permission', DEFAULT_PERMISSION);
```
Maybe you can also remove the default from the `get` function and only rely on the `||` part ?